kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js

```diff
@@ -146,102 +208,113 @@ define([
             const action = el.getAttribute('data-action'),
                 target = el.getAttribute('data-target');
 
-            if (['cancel', 'retry', 'go-to-results'].includes(action)) {
-                if (action === 'go-to-results') {
-                    // switch to results tab
-                    return jobManager.viewResults(target);
-                }
-                return jobManager[`${action}Job`](target);
+            if (!['cancel', 'retry', 'go-to-results'].includes(action)) {
+                return false;
             }
-        }
 
-        function startParamsListener(jobId) {
-            listeners[`params__${jobId}`] = bus.listen({
-                channel: {
-                    jobId: jobId,
-                },
-                key: {
-                    type: 'job-info',
-                },
-                handle: handleJobInfoUpdate,
-            });
+            const jobState = jobManager.model.getItem(`exec.jobs.byId.${target}`);
+            if (!jobState) {
+                return false;
+            }
+
+            if (action === 'go-to-results') {
+                // switch to results tab
+                return toggleTab('results');
+            }
+            if (!Jobs.canDo(action, jobState)) {
+                // make sure that the button is disabled so it cannot be clicked again
+                e.target.disabled = true;
+                return false;
+            }
+            jobManager.doJobAction(action, [target]);
+            // disable the button to prevent further clicks
+            e.target.disabled = true;
+            return true;
         }
 
-        function startStatusListener(jobId) {
-            listeners[`status__${jobId}`] = bus.listen({
-                channel: {
-                    jobId: jobId,
-                },
-                key: {
-                    type: 'job-status',
+        function setUpListeners(jobs) {
+            jobManager.addHandler('modelUpdate', {
+                dropdown: () => {
+                    dropdownWidget.updateState();
                 },
-                handle: handleJobStatusUpdate,
             });
-        }
+            const paramsRequired = [];
+            const jobIdList = [];
+            jobs.forEach((jobState) => {
+                jobIdList.push(jobState.job_id);
+                if (!jobManager.model.getItem(`exec.jobs.params.${jobState.job_id}`)) {
```

Comment:
check whether or not the job params have already been fetched and saved in the model